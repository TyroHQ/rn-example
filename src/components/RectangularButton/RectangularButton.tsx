import React from "react";
import { Text, View } from "react-native";
import styled from "styled-components";
import { Placeholder } from "../Placeholder";

const RectangularButtonContainer = styled(View)`
  background-color: lightgrey;
  padding: 10px;
`;

export class RectangularButton extends React.Component<{}> {
  static defaultProps: RectangularButton["props"];
  render() {
    return <Placeholder>Next button</Placeholder>;
  }
}
