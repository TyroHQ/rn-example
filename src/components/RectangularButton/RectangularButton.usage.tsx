import React from "react";
import { boolean, text, number } from "@storybook/addon-knobs";
import { action } from "@storybook/addon-actions";
import { RectangularButton } from "./";

export const basic = () => <RectangularButton />;
