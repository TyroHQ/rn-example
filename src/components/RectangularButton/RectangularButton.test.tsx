import { shallow } from "enzyme";
import * as usage from "./RectangularButton.usage";

describe("<RectangularButton />", () => {
  for (const example in usage) {
    it(`renders well in ${example} mode`, () => {
      const x = (usage as any)[example]();
      const wrapper = shallow(x);
      expect(wrapper).toMatchSnapshot();
    });
  }
});
