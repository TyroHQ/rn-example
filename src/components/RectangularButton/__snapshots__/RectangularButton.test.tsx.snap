// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<RectangularButton /> renders well in basic mode 1`] = `
<Placeholder
  orientation="row"
>
  Next button
</Placeholder>
`;
